libbeat/publisher/queue/memqueue/broker.go

```diff
@@ -28,7 +28,13 @@ import (
 )
 
 // Feature exposes a memory queue.
-var Feature = queue.Feature("mem", create, feature.Stable)
+var Feature = queue.Feature("mem",
+	create,
+	feature.NewDetails(
+		"Memory queue",
+		"Buffer events in memory before sending to the output.",
+		feature.Stable),
```

Comment:
I'm a little undecided here if we should rather use a struct instead of an interface+constructor.

Pro:
- Especially with go-plugins I see the advantage of using an interface for the details. One can use custom ones. Even vendoring hickups will be no problem.
- We can provide alternative constructors, e.g. filling in `feature.Stable` or have constructors adding 'class/type' information in the future. E.g. the actual type-safe feature constructor or package can provide support for constructing details via parameters or alternative functions.

Pro and cons:
- Providing a constructor will always force anyone to update feature definitions once the constructor changes. Pro: always up to date meta-data, Cons: developer adding a field must update all features.

Cons:
- if we add more information to the details, it might become more difficult to see which fields are actually set. By using a struct it's always clear if one writes `Details{FullName: ..., Doc: ...}`
- By having `Describable` and interface makes the code a little more bloated. Plus it enforces the use of the constructor, so to work around symbol-name reuse.

Reply:
I had the same indecision and a similar list.

My main reason I've opted for an interface is, is because data and need will surely change in a minor release and I wanted to make sure we reduce the changes needed by external developers.